packages/next/build/tracer.ts

```diff
@@ -2,6 +2,49 @@ import api, { Span } from '@opentelemetry/api'
 
 export const tracer = api.trace.getTracer('next', process.env.__NEXT_VERSION)
 
+const compilerStacks = new WeakMap()
+
+export function stackPush(compiler: any, spanName: string, attrs?: any): any {
```

Comment:
Using this approach allowed for easier association.  The error handling in `stackPop` down below also helped to identify when the start/stop tracing events were placed incorrectly.